platform/banana-pi/platform.c

```diff
@@ -0,0 +1,308 @@
+    if (buf[0] == NULL_CHAR)
+        return -1;
+    position = buf;
+    while(*position != NULL_CHAR)
+    {
+        if (*position == NEW_LINE)
+        {
+            *position = NULL_CHAR;
+            break;
+        }
+        position++;
+    }
+    position = strchr(buf, '=');
+    if (position == NULL)
+    {
+        wifi_hal_dbg_print("Line %d: invalid line '%s'",__LINE__, buf);
+        return -1;
+    }
+    *position = NULL_CHAR;
+    position++;
+    strncpy(out,position,strlen(position)+1);
+    return 0; 
+}
+
+int platform_pre_init()
+{
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__);    
+    return 0;
+}
+
+int platform_post_init(wifi_vap_info_map_t *vap_map)
+{
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__);    
+    system("brctl addif brlan0 wifi0");
+    system("brctl addif brlan0 wifi1");
+    return 0;
+}
+
+
+int platform_set_radio(wifi_radio_index_t index, wifi_radio_operationParam_t *operationParam)
+{
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__);    
+    return 0;
+}
+
+int platform_set_radio_pre_init(wifi_radio_index_t index, wifi_radio_operationParam_t *operationParam)
+{
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__);    
+    return 0;
+}
+
+int platform_create_vap(wifi_radio_index_t index, wifi_vap_info_map_t *map)
+{
+    char output_val[BPI_LEN_32];
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__);
+
+    if (map == NULL)
+    {
+        wifi_hal_dbg_print("%s:%d: wifi_vap_info_map_t *map is NULL \n", __func__, __LINE__);
+    }
+    for (index = 0; index < map->num_vaps; index++)
+    {
+      if (map->vap_array[index].vap_mode == wifi_vap_mode_ap)
+      {
+	//   Assigning default radius values 
+	    wifi_nvram_defaultRead("radius_s_port",output_val);
+	    map->vap_array[index].u.bss_info.security.u.radius.s_port = atoi(output_val);
+	    wifi_nvram_defaultRead("radius_s_ip",map->vap_array[index].u.bss_info.security.u.radius.s_ip);
+	    wifi_nvram_defaultRead("radius_key",map->vap_array[index].u.bss_info.security.u.radius.s_key);
+      }
+    } 
+    return 0;
+}
+
+int nvram_get_radio_enable_status(bool *radio_enable, int radio_index)
+{
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__);    
+    return 0;
+}
+
+int nvram_get_vap_enable_status(bool *vap_enable, int vap_index)
+{
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__);    
+    return 0;
+}
+
+int nvram_get_current_security_mode(wifi_security_modes_t *security_mode,int vap_index)
+{
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__);    
+    return 0;
+}
+
+int platform_get_keypassphrase_default(char *password, int vap_index)
+{
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__);  
+    /*password is not sensitive,won't grant access to real devices*/ 
+    wifi_nvram_defaultRead("bpi_wifi_password",password);
+    return 0;
+}
+
+int platform_get_ssid_default(char *ssid, int vap_index)
+{
+    wifi_hal_dbg_print("%s:%d \n",__func__,__LINE__);   
+    sprintf(ssid,"BPI_RDKB-AP%d",vap_index);
```

Comment:
Need to make sure enough space available in buffer.

Reply:
Taken care